core/jazz_splunk-kinesis-log-streamer/components/utils.js

```diff
@@ -55,47 +119,185 @@ var getSubInfo = function (message, patternStr, index) {
+                  }
+                  domainAndservice = serviceInfo;
+                  logger.debug("domainAndservice: " + domainAndservice)
+          
+                  namespace = domainAndservice.substring(0, domainAndservice.indexOf("_"));
+                  if (namespace) {
+                    data.namespace = namespace;
+                    data.service = domainAndservice.substring(namespace.length + 1);
+                  } else {
+                    data.namespace = "";
+                    data.service = domainAndservice;
+                  }
+                }
+                data.metadata.platform_log_group = payload.logGroup;
+                data.metadata.platform_log_stream = payload.logStream;
+                resolve(data);
+              } else {
+                resolve(data);
+              }
+            });
+          })
+        })
+      })
+    })
+  })
+}
+
+// Function to get service metadata using service API
+function getsServiceMetaData(config, logGroup, authToken) {
+  var serviceParts = logGroup.split('_');
   return new Promise((resolve, reject) => {
-    let data = {};
-    data.metadata = {};
-    data.asset_type = "lambda"
-    data.request_id = getInfo(payload.logEvents, global_config.PATTERNS.lambda_request_id);
-    if (data.request_id) {
-      data.provider = "aws_lambda";
-      let domainAndservice, serviceInfo;
-      data.asset_identifier = payload.logGroup.split(global_config.PATTERNS.asset_identifier_key)[1];
-      data.environment = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_environment, 2);
-      if (data.environment === "dev") {
-        let dev_environment = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_environment_dev, 2);
-        serviceInfo = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_environment_dev, 1);
-        data.environment = dev_environment;
+    var service_api_options = {
+      url: `${config.SERVICE_API_URL}${config.SERVICE_URL}?domain=${serviceParts[1]}&service=${serviceParts[2]}&environment=${serviceParts[serviceParts.length - 1]}`,
+      headers: {
+        "Content-Type": "application/json",
+        "Authorization": authToken
+      },
+      method: "GET",
+      rejectUnauthorized: false,
+      requestCert: true,
+      async: true
+    };
+
+    request(service_api_options, (error, response, body) => {
+      if (error) {
+        reject(error);
       } else {
-        serviceInfo = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_domain_service, 1);
+        if (response.statusCode && response.statusCode === 200) {
+          var responseBody = JSON.parse(body);
+          logger.debug("Response Body of Service Metadata is: " + JSON.stringify(responseBody));
+          resolve(responseBody.data.services[0])
+        } else {
+          logger.debug("Service not found for this service, domain, environment: ", JSON.stringify(service_api_options));
+          resolve([])
```

Comment:
This has to be an error - `reject(error)` instead of `resolve([])`?

Reply:
Done.